gui/src/2D/xmlLoader.ts

```diff
@@ -327,9 +331,41 @@ export class XmlLoader {
                     callback();
                 }
             }
-        }.bind(this);
+        };
 
         xhttp.open("GET", xmlFile, true);
         xhttp.send();
     }
+    /**
+     * Initiates the xml layout loading asynchronously
+     * @param xmlFile defines the xml layout to load
+     * @param rootNode defines the node / control to use as a parent for the loaded layout controls.
+     * @returns Promise
+
+     */
+    public async loadLayoutAsync(xmlFile: any, rootNode: any): Promise<boolean> {

```

Comment:
Could you instead wrap the LoadLayout function where you d expose a onError callback ? this would help factorize the code

Reply:
Yes, you're right, just did another commit.